test/api/publish.test.js

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import supertest from 'supertest';
+
+import crypto from 'crypto';
+
+import { Storage as MockStorage } from '../../__mocks__/@google-cloud/storage';
+
+import { config } from '../../src/config';
+import { createApp } from '../../src/app';
+
+beforeEach(() => MockStorage.cleanUp());
+afterEach(() => MockStorage.cleanUp());
+
+describe('publishing endpoints', () => {
+  function setup() {
+    const agent = supertest(createApp().callback());
+    return agent.post('/compressed-store').type('text');
+  }
+
+  it('uploads data all the way to google storage when using a content length', async () => {
+    const content = 'aaaa';
```

Comment:
Could you please add more inline comments to this test? I'm struggling with following along with it.